storage/block_devices_ops.go

```diff
@@ -541,21 +545,593 @@ func partitionUsingParted(bd *BlockDevice, dryRun *[]string, wholeDisk bool) err
+		return processPhysicalVolume(bd, ForceDestructive, dryRun, scans)
+	}
+
+	args := []string{"parted", parent.GetDeviceFile(), "--script", "--", "rm", strconv.FormatUint(partNum, 10)}
+
+	if dryRun == nil {
+		log.Warning("Deleting part: %s from disk: %s", bd.Name, parent.Name)
+		err = cmd.RunAndLog(args...)
+		if err != nil {
+			return errors.Wrap(err)
+		}
+	}
+
+	return nil
+}
+
+func (bd *BlockDevice) cleanUpDisk(disk string, ForceDestructive bool,
+	dryRun *DryRunType, scans *preScanResults) error {
+	var err error = nil
+
+	for _, ch := range bd.Children {
+		if err = ch.cleanUpDisk(disk, ForceDestructive, dryRun, scans); err != nil {
+			return err
+		}
+		if destroyOp, okay := bdDestroyOps[ch.Type]; okay {
+			if err = destroyOp.RundestroyCommand(ch, ForceDestructive, disk, dryRun, scans); err != nil {
+				return err
+			}
+		}
+	}
+
+	return err
+}
+
+func preScanPhysicalVolumes(bds []*BlockDevice) []*BlockDevice {
+	PhysicalVolumeFinder := func(b *BlockDevice) bool {
+		if b.FsType == "LVM2_member" {
+			return true
+		}
+		return false
+	}
+
+	return FindAllBlockDevices(bds[0], PhysicalVolumeFinder)
+}
+
+func preScanRaidMembers(bds []*BlockDevice) []*BlockDevice {
+	raidMemberFinder := func(b *BlockDevice) bool {
+		if b.FsType == "linux_raid_member" {
+			return true
+		}
+		return false
+	}
+
+	return FindAllBlockDevices(bds[0], raidMemberFinder)
+}
+
 // WritePartitionTable writes the defined partitions to the actual block device
-func (bd *BlockDevice) WritePartitionTable(legacyBios bool, wholeDisk bool, dryRun *[]string) error {
+func (bd *BlockDevice) WritePartitionTable(legacyBios bool, ForceDestructive bool, wholeDisk bool,
+	dryRun *DryRunType) error {
 	if bd.Type != BlockDeviceTypeDisk && bd.Type != BlockDeviceTypeLoop &&
 		bd.Type != BlockDeviceTypeRAID0 && bd.Type != BlockDeviceTypeRAID1 && bd.Type != BlockDeviceTypeRAID4 &&
 		bd.Type != BlockDeviceTypeRAID5 && bd.Type != BlockDeviceTypeRAID6 && bd.Type != BlockDeviceTypeRAID10 {
 		return errors.Errorf("Type is partition, disk required")
 	}
 
+	var prg progress.Progress
+
 	if wholeDisk {
-		if err := bd.removeAllLogicalVolumes(dryRun); err != nil {
+		if dryRun == nil {
+			mesg := utils.Locale.Get("Cleaning disk %s", bd.GetDeviceFile())
+			prg = progress.NewLoop(mesg)
+			log.Info("Cleaning disk %s with ForceDestructive: %v", bd.GetDeviceFile(), ForceDestructive)
+
+			if GetImpactOnOtherDisks() {
+				if !ForceDestructive {
+					prg.Failure()
+					return errors.Errorf("Cannot proceed with Installation as it impacts other disks.." +
+						" " + "Run installer with --force-destructive")
+				}
+			}
+		}
+
+		disk := bd.GetDeviceFile()
+		bds, err := getBlockDevicesLsblkJSON(disk)
+
+		scans := &preScanResults{}
+
+		scans.ScannedPvList = preScanPhysicalVolumes(bds)
+		scans.ScannedRaidList = preScanRaidMembers(bds)
+
+		if err != nil {
+			return err
+		}
+
+		if len(bds) != 1 {
+			return errors.Errorf("multiple entries found for same device")
```

Comment:
I am not sure what you mean? If you mean Localize and capitalize: I dont think this needs to be as this is never returned to the user on gui or tui... this would only show as error in log file or log trace?

Reply:
If it only ends up in the log, then localization not needed. But we generally start all log messages with a capital.
```suggestion
			return errors.Errorf("Multiple entries found for same device")
```

Reply:
will capitalize. Done